packages/spotlight/src/container.js

```diff
@@ -693,22 +693,42 @@ function getContainerNavigableElements (containerId) {
 	}
 
 	const config = getContainerConfig(containerId);
+	const {enterTo, overflow} = config;
 
-	const enterLast = config.enterTo === 'last-focused';
-	const enterDefault = config.enterTo === 'default-element';
+	const enterLast = enterTo === 'last-focused';
 	let next;
 
 	// if the container has a preferred entry point, try to find it first
 	if (enterLast) {
 		next = getContainerLastFocusedElement(containerId);
-	} else if (enterDefault) {
+	}
+
+	// try default element if last focused can't be focused
+	if (!next) {
 		next = getContainerDefaultElement(containerId);
 	}
 
-	// if there isn't a preferred entry or it wasn't found, try to find something to spot
 	if (!next) {
-		next = (!enterDefault && getContainerDefaultElement(containerId)) ||
-				getSpottableDescendants(containerId);
+		const spottables = getSpottableDescendants(containerId);
+
+		// if there isn't a preferred entry on an overflow container, try to find the first element
+		// in view
+		if (overflow) {
+			const containerRect = getContainerRect(containerId);
+			next = spottables.reduce((result, element) => {
```

Comment:
Why not use `array.find()`?  There doesn't seem to be a need to iterate over all elements once you've found the first match.

Reply:
That would work!